WordPress/gutenberg
# Font Collections: lazy load json configuration for better performance

## What?

- Restructures `WP_Font_Collection` to access font collection data using a `::get_data` method rather than public properties so that we can lazy load collection json and avoid making http requests or file reads when not needed
- Removes `wp_register_font_collection_from_json` in favor of providing a config array or a file path when using `wp_register_font_collection`
- Moves the google-fonts collection registration into a hooked function on `init` rather than the global scope

## Why?

https://github.com/WordPress/gutenberg/pull/58363 caused a performance regression. Although it added caching, making the initial request for the font collection json on collection registration causes an unnecessary http request on every page load.

## Testing Instructions

- During a regular page load on the site, or in wp-admin, use logging or otherwise to see that WP_Font_Collection::load_from_json is not called
- Load the Font Library and see that the Google Fonts collection works as expected
